earthaccess/__init__.py

```diff
@@ -86,21 +86,4 @@ def __getattr__(name):  # type: ignore
     if name not in ["__auth__", "__store__"]:
         raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
 
-    with _lock:
-        if not _auth.authenticated:
-            for strategy in ["environment", "netrc"]:
-                try:
-                    _auth.login(strategy=strategy)
-
-                    if _auth.authenticated:
-                        _store = Store(_auth)
-                        logger.debug(
-                            f"Automatic authentication with {strategy=} was successful"
-                        )
-                        break
-                except Exception as e:
-                    logger.debug(
-                        f"An error occurred during automatic authentication with {strategy=}: {str(e)}"
-                    )
-
-        return _auth if name == "__auth__" else _store
```

Comment:
It's possible some may rely on this implicit/automatic login behavior. I don't think we should necessarily keep this behavior around, but I do think we should call this out as a breaking change in the changelog if we're removing it.

And let's also update the title of this PR to mention the removal of automatic login behavior!